zeppelin-jupyter-interpreter/src/main/java/org/apache/zeppelin/jupyter/JupyterKernelInterpreter.java

```diff
@@ -183,8 +195,43 @@ public String checkKernelPrerequisite(String pythonExec) {
     return "";
   }
 
+  private void activateCondaEnv(String envName) throws IOException {
+    LOGGER.info("Activating conda env: {}", envName);
+    ByteArrayOutputStream stdout = new ByteArrayOutputStream();
+    PumpStreamHandler psh = new PumpStreamHandler(stdout);
+    try {
+      if (!new File(envName).exists()) {
+        throw new IOException("Fail to activating conda env because no environment folder: " +
+                envName);
+      }
+      File scriptFile = Files.createTempFile("zeppelin_jupyter_kernel_", ".sh").toFile();
+      try (FileWriter writer = new FileWriter(scriptFile)) {
+        IOUtils.write(String.format("chmod 777 -R %s \nsource %s/bin/activate \nconda-unpack",
+                envName, envName),
+                writer);
+      }
+      scriptFile.setExecutable(true, false);
+      scriptFile.setReadable(true, false);
+      CommandLine cmd = new CommandLine(scriptFile.getAbsolutePath());
+      DefaultExecutor executor = new DefaultExecutor();
+      executor.setStreamHandler(psh);
+      int exitCode = executor.execute(cmd);
+      if (exitCode != 0) {
+        throw new IOException("Fail to activate conda env, " + stdout.toString());
+      } else {
+        LOGGER.info("Activate conda env successfully");
+        this.condaEnv = envName;
+        this.pythonExecutable = envName + "/bin/python";
+      }
+    } catch (Exception e) {
```

Comment:
This should be changed to specific exceptions. More at [RSPEC-2221](https://rules.sonarsource.com/java/tag/error-handling/RSPEC-2221)

Reply:
Addressed